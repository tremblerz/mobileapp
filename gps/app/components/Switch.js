import React from 'react';
import { Switch as RNSwitch } from 'react-native';

import Colors from '../constants/colors';

// TODO(https://pathcheck.atlassian.net/browse/SAF-230): make theme aware
export const Switch = ({
  onValueChange,
  value = false,
  trackColor = { false: Colors.GRAY_BACKGROUND, true: Colors.SUCCESS },
  testID = 'switch',
  thumbColor = Colors.WHITE,
}) => {
  return (
    <RNSwitch
      value={value}
      onValueChange={onValueChange}
      trackColor={trackColor}
      thumbColor={thumbColor}
      testID={testID}
    />
  );
};
