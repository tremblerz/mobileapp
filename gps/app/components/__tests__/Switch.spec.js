import { render } from '@testing-library/react-native';
import React from 'react';

import { Switch } from '../Switch';

it('renders as off by default', () => {
  const { asJSON } = render(<Switch />);
  expect(asJSON()).toMatchSnapshot();
});

it('renders on', () => {
  const value = true;
  const { asJSON } = render(<Switch value={value} />);
  expect(asJSON()).toMatchSnapshot();
});
