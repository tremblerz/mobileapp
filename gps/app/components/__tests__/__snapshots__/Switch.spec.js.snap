// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as off by default 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSwitch
    accessibilityRole="button"
    onTintColor="#41dca4"
    style={
      Object {
        "height": 31,
        "width": 51,
      }
    }
    testID="switch"
    thumbTintColor="#FFF"
    tintColor="#DADADA"
    value={false}
  />
</View>
`;

exports[`renders on 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSwitch
    accessibilityRole="button"
    onTintColor="#41dca4"
    style={
      Object {
        "height": 31,
        "width": 51,
      }
    }
    testID="switch"
    thumbTintColor="#FFF"
    tintColor="#DADADA"
    value={true}
  />
</View>
`;
